samples/dialogs/dialogs.cpp

```diff
@@ -2421,16 +2421,16 @@ class TestNotificationMessageWindow : public wxFrame
 #endif
         wxSizer * const sizerTop = new wxBoxSizer(wxVERTICAL);
 
-        wxSizer* sizerText = new wxStaticBoxSizer(wxVERTICAL, this, "Notification Texts");
+        wxStaticBoxSizer* sizerText = new wxStaticBoxSizer(wxVERTICAL, this, "Notification Texts");
 
-        sizerText->Add(new wxStaticText(this, wxID_ANY, "&Title:"),
+        sizerText->Add(new wxStaticText(sizerText->GetStaticBox(), wxID_ANY, "&Title:"),
```

Comment:
I know beauty is not the most important thing in this code, but it still bothers me to repeat `sizerText->GetStaticBox()` so many times here. Could we please do something like

```suggestion
        wxStaticBox* const boxTest = sizerText->GetStaticBox();
        
        sizerText->Add(new wxStaticText(boxText, wxID_ANY, "&Title:"),
```

instead?

Reply:
Did you mean (a) just there, (b) everywhere in the dialogs sample, or (c) everywhere in both samples?

The current code in widgets sample uses `wxStaticBoxSizer` in two ways:
1. Create `wxStaticBox` first and then use it with `wxStaticBoxSizer` (often originally declared just as `wxSizer`).
2. Just create `wxStaticBoxSizer`.

When doing my changes, I opted for minimum changes to the existing code and used as the parent the existing `wxStaticBox` instance in the first case and `wxStaticBoxSizer::GetStaticBox()` in the second one.

Reply:
I think it would be nice to do it everywhere where `sizer->GetStaticBox()` is used more than 2-3 times, as IMO the code becomes even more difficult to read with these calls everywhere in it.

Reply:
I updated the code, in the spirit of the request. Hopefully it is good enough now.

Reply:
Great, thanks for the update! I'll merge this soon.